src/http/headers.cr

```diff
@@ -118,64 +162,168 @@ struct HTTP::Headers
     false
   end
 
-  def add(key, value : String)
-    check_invalid_header_content value
+  # Inserts a key value pair into the header collection and returns the value of the added key.
+  # ArgumentError will be raised if header includes invalid characters.
+  #
+  # ```
+  # headers = HTTP::Headers.new
+  # headers.add("foo", "bar") #=> HTTP::Headers{"foo" => "bar"}
+  # headers["foo"]            #=> "bar"
+  # ```
+  def add(key, value : String) : self
```

Comment:
Should probably mention that if you use `#add` with a key that already exists, it'll create an array.

https://play.crystal-lang.org/#/r/8us8

Whereas `#[]=` would override they key's value.